lib/libc/Kconfig

```diff
@@ -217,6 +217,19 @@ config MINIMAL_LIBC_TIME
 	  In order to make use of the non-reentrant gmtime(), it is necessary
 	  to set CONFIG_MINIMAL_LIBC_NON_REENTRANT_FUNCTIONS=y.
 
+# Note: inverted logic used to simplify conditionals in test_strerror.c
+config MINIMAL_LIBC_DISABLE_STRING_ERROR_TABLE
+	bool "Disable string error table for strerror() and strerror_r()"
+	help
+	  The string error table can add ~2kiB to applications when functions
+	  such as strerror(), strerror_r(), and perror() are linked-in. When
+	  those functions are not used, the string table is discarded by the
+	  linker.
+
+	  Select 'y' to reduce code size without removing calls to
+	  strerror(), strerror_r(), and perror(). When 'y' is selected
+	  strerror() and strerror_r() will produce the empty string.
```

Comment:
The error string table should be disabled by default.

Let's try to keep the minimal libc minimal as its name says. Adding string table support, although not minimal, is reasonable; but, at least, it should not be enabled by default.

Also 2KiB is by no means small when the entirety of the flash is 64KB or something.

Reply:
Sure thing - I'm fine either way. @keith-packard preferred the opposite, but since you are the libc maintainer, I will change it back.

Reply:
Done!